src/selectors/entities/roles.js

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) 2018-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import {createSelector} from 'reselect';
+
+import {getCurrentUser} from 'selectors/entities/common';
+import {getTeamMemberships} from 'selectors/entities/teams';
+import {getMyChannelMemberships} from 'selectors/entities/channels';
+
+export const getMyRoles = createSelector(
```

Comment:
Would it possibly be better to split this into `getMySystemRoles`, `getMyTeamRoles`, `getMyChannelRoles`? Just wondering because it's going to be returning new objects every time for `system`, `team` and `channel` fields even if they don't change (e.g the current user changes but not the team or channel members)

Non-blocking comment, and really I'm 0/5. Just something to think about

Reply:
Let me jump here saying that i’m actually 3/5 on splitting the roles as much as possible to avoid a new object cause i’m assuming those objects are going to be used everywhere unless i’m totally wrong and is a one time thing

Reply:
Completely agree, actually, the getMyXPerms and haveIXPerm selectors depends on that, so will improve that too using only the required ones.

Reply:
Fixed too